### PR TITLE
Fix breaking changes for `runtime.Config` usage

### PR DESCRIPTION
**Pull request by m-Peter**
## Description

- Remove `CoverageReportedRuntime` and pass `CoverageReport` in `runtime.Config`.
- Remove also the `SetCoverageReport` method and provide instead a function `Option`, namely `WithCoverageReport`.
______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: blockchain.go [hunk 1]
```diff
@@ -1620,13 +1613,13 @@ func (b *Blockchain) EndDebugging() {
...
 
 func (b *Blockchain) CoverageReport() *runtime.CoverageReport {
-	return b.coverageReportedRuntime.CoverageReport
+	return b.coverageReport
 }
 
 func (b *Blockchain) SetCoverageReport(coverageReport *runtime.CoverageReport) {
-	b.coverageReportedRuntime.CoverageReport = coverageReport
+	b.coverageReport = coverageReport
 }
```
Comment (bluesign) [inline blockchain.go:1621]: I think this is pointless now, as we move with config

Comment (bluesign) [inline blockchain.go:1621]: I think this is pointless now, as we move with config

**Comment by turbolent** [comment]
Nice, thank you with starting the update here already.

We'll also need to update to the latest flow-go version, the update is in progress here: https://github.com/onflow/flow-go/pull/4390

**Comment by m-Peter** [comment]
I just opened this as a starting point. It's not done yet, we still need to fix the breaking changes from `flow-go`. I will keep an eye for a new version of `flow-go` and bump the version here. And I certainly need to check it end-to-end, as soon as it builds.

**Comment by turbolent** [comment]
@m-Peter Opened #393 for the update. Could you please rebase your changes on top of it?

File: blockchain.go (hunk 1 above)
Comment (m-Peter) [reply blockchain.go]: Good catch! I've removed it entirely.

**Comment by m-Peter** [comment]
@turbolent Rebased on master, to include the latest changes, and then on top of the branch you mentioned. Along the way, I did some tech debt refactoring.

**Pull request merged**